### PR TITLE
rdpsnd: Enhance server implementation

### PR DESCRIPTION
**Pull request by pnowack**
Quoting the commit message here:

```
The current server sided channel handling of RDPSND/AUDIO_PLAYBACK_DVC
is currently very constrained.
So, solve this. This means:

- Add the missing Training/Training Confirm PDUs
- Stop overriding the average bytes per second values, when submitting
  the audio formats, as this currently makes the usage of codecs
  impossible
- Add a way to send the server formats manually again, to be able to
  restart the protocol after a Close PDU was sent
- Add a way to send already encoded audio data to let server
  implementations to take care of the encoding process and to set
  custom audio timestamps for the Video Optimized Remoting channel
- Add public attributes to let server implementations know the initial
  volume and pitch values
- Add public attribute to let server implementations know the quality
  mode setting
```

I need this MR and the backport of this MR for https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/merge_requests/104

Note: This MR is a forward port of https://github.com/pnowack/FreeRDP/commits/enhance-rdpsnd. I tested the version for the stable-2.0 branch, but cannot test the branch for master.
In rdpsnd_main.c was only a tiny merge conflict, where the dynamic-static channel handle retrieval happens.
There were quite more merge conflicts in the public header file, but these should be trivial to review.

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: channels/rdpsnd/server/rdpsnd_main.c [hunk 1]
```diff
@@ -978,40 +1034,39 @@ RdpsndServerContext* rdpsnd_server_context_new(HANDLE vcm)
...
 	priv->ownThread = TRUE;
 	return context;
-out_free_dsp:
-	freerdp_dsp_context_free(priv->dsp_context);
-out_free_priv:
-	free(context->priv);
-out_free:
-	free(context);
+fail:
+	rdpsnd_server_context_new(context);
```
Comment (hardening) [inline channels/rdpsnd/server/rdpsnd_main.c:1061]: `rdpsnd_server_context_free(context);` ?

File: include/freerdp/codec/audio.h [hunk 2]
```diff
@@ -42,23 +42,23 @@ struct AUDIO_FORMAT
 };
 typedef struct AUDIO_FORMAT AUDIO_FORMAT;
 
-#define SNDC_CLOSE 0x01
```
Comment (hardening) [inline include/freerdp/codec/audio.h]: Purely cosmetic but it was nice to have these constants in hexa to match them in the spec (where they are in hexa)

File: channels/rdpsnd/server/rdpsnd_main.c (hunk 1 above)
Comment (akallabeth) [reply channels/rdpsnd/server/rdpsnd_main.c:1061]: @hardening damn, good catch, did want to write `_free`

File: include/freerdp/codec/audio.h (hunk 2 above)
Comment (akallabeth) [reply include/freerdp/codec/audio.h]: true, but trying to keep the changes minimal (apart from sound channel here)

**Comment by akallabeth** [comment]
@freerdp-bot test

**Review by akallabeth — APPROVED** [review]

**Pull request merged**